## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 510658`
Commit: [`e4e10af5627b5aab82b7d301aa84292d2d65fb75`](https://github.com/NixOS/nixpkgs/commit/e4e10af5627b5aab82b7d301aa84292d2d65fb75) ([subsequent changes](https://github.com/NixOS/nixpkgs/compare/e4e10af5627b5aab82b7d301aa84292d2d65fb75..pull/510658/head))
Merge: [`e451f6ea275ba0441ec3d278ac841f0138141acf`](https://github.com/NixOS/nixpkgs/commit/e451f6ea275ba0441ec3d278ac841f0138141acf)

Logs: https://github.com/kybe236/nixpkgs-review-gha/actions/runs/24524089673


---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>fishPlugins.wakatime-fish</li>
    <li>wakatime-cli</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>fishPlugins.wakatime-fish</li>
    <li>wakatime-cli</li>
  </ul>
</details>

---
### `x86_64-darwin` (sandbox = relaxed)
<details>
  <summary>:x: 2 packages failed to build:</summary>
  <ul>
    <li>fishPlugins.wakatime-fish</li>
    <li>wakatime-cli</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-darwin`</summary>
<details>
<summary>wakatime-cli</summary>
<pre>ok  	github.com/wakatime/wakatime-cli/pkg/log/setup	0.580s
ok  	github.com/wakatime/wakatime-cli/pkg/offline	1.030s
ok  	github.com/wakatime/wakatime-cli/pkg/output	0.447s
{&quot;level&quot;:&quot;warn&quot;,&quot;now&quot;:&quot;2026-04-16T17:30:22Z&quot;,&quot;caller&quot;:&quot;params/params.go:1470&quot;,&quot;func&quot;:&quot;github.com/wakatime/wakatime-cli/pkg/params.parseBoolOrRegexList&quot;,&quot;message&quot;:&quot;failed to compile regex pattern \&quot;(?i)[\&quot;, it will be ignored&quot;,&quot;version&quot;:&quot;2.3.0&quot;,&quot;os/arch&quot;:&quot;unset/unset&quot;}
{&quot;level&quot;:&quot;warn&quot;,&quot;now&quot;:&quot;2026-04-16T17:30:22Z&quot;,&quot;caller&quot;:&quot;params/params.go:1470&quot;,&quot;func&quot;:&quot;github.com/wakatime/wakatime-cli/pkg/params.parseBoolOrRegexList&quot;,&quot;message&quot;:&quot;failed to compile regex pattern \&quot;(?i)[\&quot;, it will be ignored&quot;,&quot;version&quot;:&quot;2.3.0&quot;,&quot;os/arch&quot;:&quot;unset/unset&quot;}
printf: warning: ignoring excess arguments, starting with &#x27;&gt;&amp;2&#x27;
--- FAIL: TestLoadParams_APIKey_FromVault_Err_Darwin (0.04s)
    params_test.go:2269: 
        	Error Trace:	/nix/build/nix-26343-2823830582/source/pkg/params/params_test.go:2269
        	Error:      	Error message not equal:
        	            	expected: &quot;failed to read api key from vault: exit status 1&quot;
        	            	actual  : &quot;invalid api key format&quot;
        	Test:       	TestLoadParams_APIKey_FromVault_Err_Darwin
{&quot;level&quot;:&quot;warn&quot;,&quot;now&quot;:&quot;2026-04-16T17:30:22Z&quot;,&quot;caller&quot;:&quot;params/params.go:474&quot;,&quot;func&quot;:&quot;github.com/wakatime/wakatime-cli/pkg/params.loadBackoffParams&quot;,&quot;message&quot;:&quot;failed to parse backoff_at: parsing time \&quot;2021-08-30\&quot; as \&quot;2006-01-02T15:04:05Z07:00\&quot;: cannot parse \&quot;\&quot; as \&quot;T\&quot;&quot;,&quot;version&quot;:&quot;2.3.0&quot;,&quot;os/arch&quot;:&quot;unset/unset&quot;}
{&quot;level&quot;:&quot;warn&quot;,&quot;now&quot;:&quot;2026-04-16T17:30:22Z&quot;,&quot;caller&quot;:&quot;params/params.go:895&quot;,&quot;func&quot;:&quot;github.com/wakatime/wakatime-cli/pkg/params.LoadOfflineParams&quot;,&quot;message&quot;:&quot;argument --heartbeat-rate-limit-seconds must be zero or a positive integer number, got -1&quot;,&quot;version&quot;:&quot;2.3.0&quot;,&quot;os/arch&quot;:&quot;unset/unset&quot;}
{&quot;level&quot;:&quot;warn&quot;,&quot;now&quot;:&quot;2026-04-16T17:30:22Z&quot;,&quot;caller&quot;:&quot;params/params.go:926&quot;,&quot;func&quot;:&quot;github.com/wakatime/wakatime-cli/pkg/params.LoadOfflineParams&quot;,&quot;message&quot;:&quot;failed to parse heartbeats_last_sent_at: parsing time \&quot;2021-08-30\&quot; as \&quot;2006-01-02T15:04:05Z07:00\&quot;: cannot parse \&quot;\&quot; as \&quot;T\&quot;&quot;,&quot;version&quot;:&quot;2.3.0&quot;,&quot;os/arch&quot;:&quot;unset/unset&quot;}
{&quot;level&quot;:&quot;warn&quot;,&quot;now&quot;:&quot;2026-04-16T17:30:22Z&quot;,&quot;caller&quot;:&quot;params/params.go:910&quot;,&quot;func&quot;:&quot;github.com/wakatime/wakatime-cli/pkg/params.LoadOfflineParams&quot;,&quot;message&quot;:&quot;argument --sync-offline-activity must be zero or a positive integer number, got -1&quot;,&quot;version&quot;:&quot;2.3.0&quot;,&quot;os/arch&quot;:&quot;unset/unset&quot;}
FAIL
FAIL	github.com/wakatime/wakatime-cli/pkg/params	0.917s
FAIL</pre>
</details>
</details>

---
### `aarch64-darwin` (sandbox = relaxed)
<details>
  <summary>:x: 2 packages failed to build:</summary>
  <ul>
    <li>fishPlugins.wakatime-fish</li>
    <li>wakatime-cli</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `aarch64-darwin`</summary>
<details>
<summary>wakatime-cli</summary>
<pre>ok  	github.com/wakatime/wakatime-cli/pkg/log/setup	0.008s
ok  	github.com/wakatime/wakatime-cli/pkg/offline	0.384s
ok  	github.com/wakatime/wakatime-cli/pkg/output	0.009s
{&quot;level&quot;:&quot;warn&quot;,&quot;now&quot;:&quot;2026-04-16T17:26:37Z&quot;,&quot;caller&quot;:&quot;params/params.go:1470&quot;,&quot;func&quot;:&quot;github.com/wakatime/wakatime-cli/pkg/params.parseBoolOrRegexList&quot;,&quot;message&quot;:&quot;failed to compile regex pattern \&quot;(?i)[\&quot;, it will be ignored&quot;,&quot;version&quot;:&quot;2.3.0&quot;,&quot;os/arch&quot;:&quot;unset/unset&quot;}
{&quot;level&quot;:&quot;warn&quot;,&quot;now&quot;:&quot;2026-04-16T17:26:37Z&quot;,&quot;caller&quot;:&quot;params/params.go:1470&quot;,&quot;func&quot;:&quot;github.com/wakatime/wakatime-cli/pkg/params.parseBoolOrRegexList&quot;,&quot;message&quot;:&quot;failed to compile regex pattern \&quot;(?i)[\&quot;, it will be ignored&quot;,&quot;version&quot;:&quot;2.3.0&quot;,&quot;os/arch&quot;:&quot;unset/unset&quot;}
printf: warning: ignoring excess arguments, starting with &#x27;&gt;&amp;2&#x27;
--- FAIL: TestLoadParams_APIKey_FromVault_Err_Darwin (0.01s)
    params_test.go:2269: 
        	Error Trace:	/nix/build/nix-11343-3187867855/source/pkg/params/params_test.go:2269
        	Error:      	Error message not equal:
        	            	expected: &quot;failed to read api key from vault: exit status 1&quot;
        	            	actual  : &quot;invalid api key format&quot;
        	Test:       	TestLoadParams_APIKey_FromVault_Err_Darwin
{&quot;level&quot;:&quot;warn&quot;,&quot;now&quot;:&quot;2026-04-16T17:26:37Z&quot;,&quot;caller&quot;:&quot;params/params.go:474&quot;,&quot;func&quot;:&quot;github.com/wakatime/wakatime-cli/pkg/params.loadBackoffParams&quot;,&quot;message&quot;:&quot;failed to parse backoff_at: parsing time \&quot;2021-08-30\&quot; as \&quot;2006-01-02T15:04:05Z07:00\&quot;: cannot parse \&quot;\&quot; as \&quot;T\&quot;&quot;,&quot;version&quot;:&quot;2.3.0&quot;,&quot;os/arch&quot;:&quot;unset/unset&quot;}
{&quot;level&quot;:&quot;warn&quot;,&quot;now&quot;:&quot;2026-04-16T17:26:37Z&quot;,&quot;caller&quot;:&quot;params/params.go:895&quot;,&quot;func&quot;:&quot;github.com/wakatime/wakatime-cli/pkg/params.LoadOfflineParams&quot;,&quot;message&quot;:&quot;argument --heartbeat-rate-limit-seconds must be zero or a positive integer number, got -1&quot;,&quot;version&quot;:&quot;2.3.0&quot;,&quot;os/arch&quot;:&quot;unset/unset&quot;}
{&quot;level&quot;:&quot;warn&quot;,&quot;now&quot;:&quot;2026-04-16T17:26:37Z&quot;,&quot;caller&quot;:&quot;params/params.go:926&quot;,&quot;func&quot;:&quot;github.com/wakatime/wakatime-cli/pkg/params.LoadOfflineParams&quot;,&quot;message&quot;:&quot;failed to parse heartbeats_last_sent_at: parsing time \&quot;2021-08-30\&quot; as \&quot;2006-01-02T15:04:05Z07:00\&quot;: cannot parse \&quot;\&quot; as \&quot;T\&quot;&quot;,&quot;version&quot;:&quot;2.3.0&quot;,&quot;os/arch&quot;:&quot;unset/unset&quot;}
{&quot;level&quot;:&quot;warn&quot;,&quot;now&quot;:&quot;2026-04-16T17:26:37Z&quot;,&quot;caller&quot;:&quot;params/params.go:910&quot;,&quot;func&quot;:&quot;github.com/wakatime/wakatime-cli/pkg/params.LoadOfflineParams&quot;,&quot;message&quot;:&quot;argument --sync-offline-activity must be zero or a positive integer number, got -1&quot;,&quot;version&quot;:&quot;2.3.0&quot;,&quot;os/arch&quot;:&quot;unset/unset&quot;}
FAIL
FAIL	github.com/wakatime/wakatime-cli/pkg/params	0.043s
FAIL</pre>
</details>
</details>

--- a/pkgs/by-name/wa/wakatime-cli/package.nix
+++ b/pkgs/by-name/wa/wakatime-cli/package.nix
@@ -1,6 +1,5 @@
 {
   lib,
-  stdenv,
   buildGoModule,
   fetchFromGitHub,
   testers,
@@ -10,13 +9,13 @@
 
 buildGoModule (finalAttrs: {
   pname = "wakatime-cli";
-  version = "2.2.0";
+  version = "2.3.0";
 
   src = fetchFromGitHub {
     owner = "wakatime";
     repo = "wakatime-cli";
     tag = "v${finalAttrs.version}";
-    hash = "sha256-TOdAcT+IS6XONIfkASRvd7oTSSW5VX8w5q5ASvJlQb8=";
+    hash = "sha256-t0HKRN7qguE+zhyI3fbBtuhpDAPZrbMS9wa8crQkLhc=";
   };
 
   vendorHash = "sha256-HngszNLX2b2EVvh8ovouIEvjBOJL1jA5AhA6Y11ke9Y=";
@@ -27,23 +26,14 @@ buildGoModule (finalAttrs: {
     "-X github.com/wakatime/wakatime-cli/pkg/version.Version=${finalAttrs.version}"
   ];
 
-  # dial tcp 127.0.0.1:51272: connect: operation not permitted
-  # and goroutine 33 [IO wait, 10 minutes] on darwin
-  doCheck = !stdenv.hostPlatform.isDarwin;
+  __darwinAllowLocalNetworking = true;
 
   nativeCheckInputs = [ writableTmpDirAsHomeHook ];
 
   checkFlags =
     let
       skippedTests = [
-        # Tests requiring network
-        "TestFileExperts"
-        "TestSendHeartbeats"
-        "TestSendHeartbeats_ExtraHeartbeats"
-        "TestSendHeartbeats_IsUnsavedEntity"
-        "TestSendHeartbeats_NonExistingExtraHeartbeatsEntity"
-        "TestSendHeartbeats_ExtraHeartbeatsIsUnsavedEntity"
-        "TestFileExperts_Err(Auth|Api|BadRequest)"
+        "TestLoadParams_APIKey_FromVault_Err_Darwin"
       ];
     in
     [ "-skip=^${builtins.concatStringsSep "$|^" skippedTests}$" ];

